src/librustc_trans/trans/debuginfo.rs

```diff
@@ -1156,12 +1156,42 @@ pub fn get_cleanup_debug_loc_for_ast_node<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
         }
     }
 
-    NodeInfo {
+    NodeIdAndSpan {
         id: node_id,
         span: cleanup_span
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum DebugLocation {
```

Comment:
Sure, either way is fine. I think I prefer the namespaced way and don't find it too heavyweight (I've been using namespaced enums a lot though, perhaps for others it looks worse), but not strongly.